src/go/rdctl/cmd/root.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright © 2022 SUSE LLC
+
+*/
+
+// Package cmd is the main package for this CLI
+package cmd
+
+import (
+  "encoding/json"
+  "fmt"
+  "io/ioutil"
+  "log"
+  "net/http"
+  "os"
+  "path/filepath"
+  "strconv"
+
+  "github.com/spf13/cobra"
+)
+
+var (
+  // Used for flags
+  configPath  string
+  user string
+  port string
+  password string
+)
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "rdctl",
+	Short: "A CLI for Rancher Desktop",
+	Long: `rdctl can be used to drive Rancher Desktop in headless mode.
+Supported commands include getting the current preferences,
+changing settings (with an automatic restart when needed,
+and shutting Rancher Desktop down.`,
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	err := rootCmd.Execute()
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func init() {
+  cobra.OnInitialize(initConfig)
+
+  rootCmd.PersistentFlags().StringVar(&configPath, "config-path", "", "config file (default is $APPHOME/rancher-desktop/rd-engine.json)")
```

Comment:
The default value is unclear (what is `$APPHOME`)? I understand what it is, but you may need either platform-specific code to convert it to a real path the user can understand.

Reply:
Done by calculating `defaultConfigPath` and using that.